carbonmark-data/components/charts/helpers/DataTable/Table/index.tsx

```diff
@@ -1,93 +1,28 @@
 import {
   ConfigurationKey,
-  fetchData,
-  getCardRenderer,
-  getColumns,
+  getDesktopRenderer,
+  getMobileRenderer,
 } from "../configurations";
-import { CardRenderer, Columns } from "../configurations/types";
+import { DataRenderer } from "../configurations/types";
 import styles from "./styles.module.scss";
 
-/** A Client Component (wannabe Server Component) that renders the actual Table
- * configurationKey: Table configuration key
- * page: the page of the dataset to render
- */
+/** A Server component that renders a single page Data Table */
 export default function Table<RI>(props: {
   configurationKey: ConfigurationKey;
-  page: number;
-}) {
-  const locale = currentLocale();
-  const [data, setData] = useState<PaginatedResponse<RI>>(
-    EMPTY_PAGINATED_RESPONSE
-  );
-  useEffect(() => {
-    fetchData(props.configurationKey, props.page).then((data) => {
-      /* FIXME: Unfortunately we have to hardcast this */
-      setData(data as PaginatedResponse<RI>);
-    });
-  }, [props.page]);
-  const columns = getColumns(props.configurationKey, locale) as Columns<RI>;
-  const cardRenderer = getCardRenderer(
-    props.configurationKey
-  ) as CardRenderer<RI>;
-
-  return (
-    <>
-      <TableLayout data={data} columns={columns}></TableLayout>
-      <CardsLayout data={data} cardRenderer={cardRenderer}></CardsLayout>
-    </>
-  );
-}
-
-/**
- * Display data as a table
- */
-function TableLayout<RI>(props: {
   data: PaginatedResponse<RI>;
-  columns: Columns<RI>;
-}) {
-  const columnKeys = Object.keys(props.columns);
-  const columns = props.columns;
-  return (
-    <table className={styles.table}>
-      <thead>
-        <tr className={styles.header}>
-          {columnKeys.map((key) => (
-            <th key={key} className={columns[key].cellStyle}>
-              {columns[key].header}
-            </th>
-          ))}
-        </tr>
-      </thead>
-      <tbody>
-        {props.data.items.map((item, index) => (
-          <tr key={index} className={styles.row}>
-            {columnKeys.map((key) => (
-              <td key={key} className={columns[key].cellStyle}>
-                {columns[key].formatter(item[columns[key].dataKey] as never)}
-              </td>
-            ))}
-          </tr>
-        ))}
-      </tbody>
-    </table>
-  );
-}
-
-/**
- * Displays data as cards
- */
-function CardsLayout<RI>(props: {
-  data: PaginatedResponse<RI>;
-  cardRenderer: CardRenderer<RI>;
 }) {
+  const desktopRenderer = getDesktopRenderer(
+    props.configurationKey
+  ) as DataRenderer<RI>;
+  const mobileRenderer = getMobileRenderer(
+    props.configurationKey
+  ) as DataRenderer<RI>;
+  const data = props.data;
   return (
-    <div className={styles.cards}>
-      {props.data.items.map((item) => props.cardRenderer({ item }))}
+    <div className={styles.table}>
+      <div className={styles.desktopOnly}>{desktopRenderer({ data })}</div>
+      <div className={styles.mobileOnly}>{mobileRenderer({ data })}</div>
```

Comment:
Hmm.. I cannot see the flickering.
This should not be flicker on Vercel deployments because CSS is supposedly loaded on the browser before the HTML body.
Can you try to reproduce here? https://carbonmark-data-oivfeoqx0-klimadao.vercel.app/fr/retirement_trends

Reply:
Yeah I couldn't see it